[vanguard] Add property "description" to POI

peoplepeople000 opened the pull request:
# Description
Add the property "description" to POI and make a input box to let user input the description of the reportData. Also, in the cluster drawer, we can see the description of each reportData.

# Changes
![截圖 2024-03-20 晚上9 30 21](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/69355920-d04e-4542-80d3-ee48ff77fc6d)
![截圖 2024-03-20 晚上9 33 13](https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/5acb2aea-29fa-4cc0-a337-dd1a74703ae4)




# Notes
Because there's no corresponding icon, so I used other icon instead.

# Checklist

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] Any changes to strings have been published to our translation tool
- [x] I conducted basic QA to assure all features are working
- [x] I requested code review from other team members

# Resolved Issues
#92

linyc0817 commented on src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx:
```diff
@@ -464,6 +372,45 @@ const StatusValueSelect: React.FC = () => {
+  const reportData = useSelector((state: IRootState) => state.report.data);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value;
+    dispatch(
+      updateAddReportData({
+        target: {
+          ...reportData.target,
+          description: newValue,
+        },
+      }),
+    );
```
It is better to pass the value directly. No need to create a new parameter.

peoplepeople000 replied:
I removed the newValue parameter, and passed e.target.value to description directly.

linyc0817 commented on src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx:
```diff
@@ -604,7 +551,7 @@ const AddReportDrawerContent: React.FC = () => {
                 alt="status value"
```
Alt means what this section means, so `status value` might not be suitable here.

peoplepeople000 replied:
I changed the alt to "description".

linyc0817 commented on src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx:
```diff
@@ -604,7 +551,7 @@ const AddReportDrawerContent: React.FC = () => {
               />
             </div>
-            <StatusValueSelect />
+            <AddDescription />
```
Follow the naming rules of previous components. e.g. StatusValueSelect
So you might use `StatusDescriptionAdd`? Or maybe you can come up with a better name.

peoplepeople000 replied:
I changed the name of the component to StatusDescriptionAdd.

linyc0817 commented on src/components/Drawer/EditReportDrawer/EditReportDrawerContent.tsx:
```diff
@@ -69,6 +68,45 @@ const StatusValueSelect: React.FC = () => {
+  const reportData = useSelector((state: IRootState) => state.report.data);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value;
+    dispatch(
+      updateAddReportData({
+        target: {
+          ...reportData.target,
+          description: newValue,
+        },
+      }),
+    );
```
Same as above, pass value directly

peoplepeople000 replied:
I removed the newValue parameter, and passed e.target.value to description directly.

linyc0817 commented on src/components/Drawer/EditReportDrawer/EditReportDrawerContent.tsx:
```diff
@@ -199,6 +221,17 @@ const EditReportDrawerContent: React.FC = () => {
+              alt="status value"
+            />
+          </div>
+          <EditDescription />
```
Same as above, change the component name

peoplepeople000 replied:
I changed the name of the component to StatusDescriptionEdit.

linyc0817 reviewed (changes requested):
Looks good overall.
Please help to check & fix some points.

linyc0817 reviewed (approved):
Well done.
But when i run the code, i found this warning `A component changed from controlled to uncontrolled.` at the console, can you try to fix it. (Seems like it may cause by NextUI )
<img width="1309" alt="截圖 2024-04-06 上午12 12 03" src="https://github.com/CAMPUS-NYCU/smart-campus/assets/32840701/df086135-b7c8-4949-b7cd-45987b7f544b">

linyc0817 commented:
## Bug reproduce
https://github.com/CAMPUS-NYCU/smart-campus/assets/32840701/a4daea83-76e7-4b6c-a059-b74ab1d8f909

- May become error sometimes
<img width="1336" alt="截圖 2024-04-08 上午12 11 40" src="https://github.com/CAMPUS-NYCU/smart-campus/assets/32840701/a3746067-2570-4506-bdb2-0059a779b6e3">

peoplepeople000 commented:
<img width="889" alt="截圖 2024-04-17 晚上8 41 25" src="https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/95802616-7703-42bc-b5e2-ea363f440687">
I added "" in row 101 to try to fix the bug.

<img width="821" alt="截圖 2024-04-17 晚上8 43 37" src="https://github.com/CAMPUS-NYCU/smart-campus/assets/107465871/cf2dbeb2-68da-45be-bb98-eb7305eacd61">
I added "" in row 405 to try to fix the bug.

Because the error did not appear on my device. So I'm not pretty sure that if the error was fixed or not.

linyc0817 reviewed (approved):
Well done
